src/qibosoq/rfsoc_server.py

```diff
@@ -15,6 +15,11 @@
 
 from qibosoq.qick_programs import ExecutePulseSequence, ExecuteSingleSweep
 
+# initialize QickSoc object (firmware and clocks)
+global_soc = QickSoc()
```

Comment:
Why do you need a global object?

In principle, you'd only need a single global object in your application, and that will be the handler.

Even more, if you use the `serve()` function, even that object will be local to `serve()`, and only the function call will be done in the global scope.

Reply:
However, if you need it, independently of the handler, instantiate it inside `serve()`.

Reply:
Everytime QickSoc gets initialized it slowly re-loads the firmware, resetting clocks and changing the phase between DACs and ADCs, so it's important to not do it.

I understand that is better, in any case, to have it inside the serve function, but if I try I get an error:
```
File "/home/xilinx/qibosoq/src/qibosoq/rfsoc_server.py", line 45, in handle
    program = ExecutePulseSequence(global_soc, data["cfg"], data["sequence"], data["qubits"])
NameError: name 'global_soc' is not defined
```

At first I though to pass the QickSoc object as an attribute of MyTCPHandler/TCPServer, rewriting __init__ or adding a function, but I couldn't really do it and opted for the global

Reply:
> The problem with this was that I do not have control on the MyTCPHandler object initialization. This is done inside the TCPServer class ([python/cpython@`7f70f1c`/Lib/socketserver.py#L361](https://github.com/python/cpython/blob/7f70f1c5cc964f64a89b3536b1bc96c3ecb6cb24/Lib/socketserver.py#L361)) and I didn't want to change that object at all (I could do it, but it seems a bit too much).

The suggestion above is really to extend the initialization of `TCPHandler` in `MyTCPHandler`, adding a simple step before, i.e. `soc` assignment.

With:
```py
class MyTCPHandler(BaseRequestHandler):
    def __init__(self, soc, *args, **kwargs):
        self.soc = soc
        super().__init__(*args, **kwargs)
```
you're simply saying:
- get all arguments that are passed
- the first one is special: assign to an attribute, and strip from the list
- pass down the rest of the arguments and execute the parent `__init__`, as if I did nothing and just received the same arguments but `soc`

This does not require patching CPython, it is just standard practice in subclassing. If you didn't want to modify the original class at all, you should not use subclassing, but composition:
```py
class MyTCPHandler:
    def __init__(self, soc, *args, **kwargs):
        self.soc = soc
        self.handler = BaseRequestHandler(*args, **kwargs)
```

But given that it is named `BaseRequestHandler` I really believe it is intentionally made for subclassing, and extending `__init__` has nothing special.

Reply:
Ok, if you're not constructing it, then we can think about setting after initialization.

I.e.:
```py
class MyTCPHandler(BaseRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.soc = None
```
and then find a way to set `handler.soc = soc` afterward.

I need to have a look to the server, to see whether it exposes the handler, and how it gets access to the class.